### ISSUE TITLE
attention_mask is getting set as `torch.ones`

### ISSUE THREAD

**Issue by mikkelfo** [FGA-DIKU/EHR#164]
Currently, we pass in an attention_mask that includes padding https://github.com/FGA-DIKU/EHR/blob/cceb21611e1a001ea3278c4cab2a97fd582b767e/corebehrt/modules/model/model.py#L22 This makes the new ModernBERT unpadding approach unused, which should give a slowdown

**Comment by kirilklein** [comment]
For the spedup to take effect, it needs FA2 for unpadding.
